crates/ethrpc/src/buffered.rs

```diff
@@ -84,26 +85,57 @@ where
         tokio::task::spawn(batched_for_each(config, calls, move |batch| {
             let inner = inner.clone();
             async move {
-                let (mut requests, mut senders): (Vec<_>, Vec<_>) = batch
-                    .into_iter()
-                    .filter(|(_, _, sender)| !sender.is_canceled())
-                    .map(|(id, request, sender)| ((id, request), sender))
-                    .unzip();
+                let (mut requests, mut trace_ids, mut senders): (Vec<_>, Vec<_>, Vec<_>) =
+                    itertools::multiunzip(
+                        batch
+                            .into_iter()
+                            .filter(|(_, _, _, sender)| !sender.is_canceled())
+                            .map(|(id, request, trace_id, sender)| {
+                                ((id, request), trace_id, sender)
+                            }),
+                    );
                 match requests.len() {
                     0 => (),
                     1 => {
-                        let ((id, request), sender) = (requests.remove(0), senders.remove(0));
-                        let result = inner.send(id, request).await;
-                        let _ = sender.send(result);
+                        let ((id, request), trace_id, sender) =
+                            (requests.remove(0), trace_ids.remove(0), senders.remove(0));
+                        tokio::spawn(async move {
```

Comment:
Why do you spawn new tasks here and in L128 when you are already inside an async block and can call directly the `await` statement? the operations which are doing in the new tasks don't seem to be blocking

Reply:
Here is a long read explaining my concern: https://github.com/cowprotocol/services/pull/2750#discussion_r1620041025. Additional spawns were removed after all.

Reply:
Thanks you for pointing out to the comment! :pray: